dipy/align/crosscorr.pyx

```diff
@@ -61,91 +160,134 @@ def precompute_cc_factors_3d(floating[:, :, :] static, floating[:, :, :] moving,
 
     References
     ----------
-    [Avants08] Avants, B. B., Epstein, C. L., Grossman, M., & Gee, J. C. (2008)
-               Symmetric Diffeomorphic Image Registration with
-               Cross-Correlation: Evaluating Automated Labeling of Elderly and
-               Neurodegenerative Brain, Med Image Anal. 12(1), 26-41.
-    [Avants11] Avants, B. B., Tustison, N., & Song, G. (2011).
-               Advanced Normalization Tools ( ANTS ), 1-35.
+    .. [Ocegueda2016] Ocegueda, O., Dalmau, O., Garyfallidis, E., Descoteaux,
+        M., & Rivera, M. (2016). On the computation of integrals over
+        fixed-size rectangles of arbitrary dimension, Pattern Recognition
+        Letters. doi:10.1016/j.patrec.2016.05.008
+    .. [Avants2008] Avants, B. B., Epstein, C. L., Grossman, M., & Gee, J. C.
+        (2008). Symmetric Diffeomorphic Image Registration with
+        Cross-Correlation: Evaluating Automated Labeling of Elderly and
+        Neurodegenerative Brain, Med Image Anal. 12(1), 26-41.
+    .. [Avants2011] Avants, B. B., Tustison, N., & Song, G. (2011). Advanced
+        Normalization Tools ( ANTS ), 1-35.
     """
     cdef:
-        cnp.npy_intp side = 2 * radius + 1
         cnp.npy_intp ns = static.shape[0]
         cnp.npy_intp nr = static.shape[1]
         cnp.npy_intp nc = static.shape[2]
-        cnp.npy_intp s, r, c, k, i, j, t, q, qq, firstc, lastc, firstr, lastr
-        double Imean, Jmean
+        cnp.npy_intp side = 2 * radius + 1
+        cnp.npy_intp firstc, lastc, firstr, lastr, firsts, lasts
+        cnp.npy_intp s, r, c, it, sides, sider, sidec
+        double cnt
+        cnp.npy_intp ssss, sss, ss, rr, cc, prev_ss, prev_rr, prev_cc
+        double Imean, Jmean, IJprods, Isq, Jsq
+        double[:, :, :, :] temp = np.zeros((2, nr, nc, 5), dtype=np.float64)
```

Comment:
`CNT` refers to the index of an array element containing the `count` of voxels within a window. For example when we compute the mean of the `I` image values within the window:
`Imean = sums[SI] / sums[CNT]`
It happens to be equal to 5, but I think it's better to be explicit here.